### PR TITLE
Add checkerboard background for the trim view

### LINKED ISSUE #113: Replace black background on trimming view with a checkerboard NSColor
[Issuehunt badges](https://issuehunt.io/r/sindresorhus/gifski-app/issues/113)

Currently we display a black background for the trimming player. While it’s a built-in behavior, I’m afraid there is no easy way of improving it. One possible solution would be to check how the player is constructed and then update the correct subview’s background color. Similar to how we remove trimming buttons.

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $60.00)

- [issuehunt issuehunt](https://issuehunt.io/u/issuehunt) ($60.00)
### Submitted pull Requests
- [#140 Add Checkerboard background for Trimming view](https://issuehunt.io/r/sindresorhus/gifski-app/pull/140)
---

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/gifski-app/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
---
IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### PR DESCRIPTION
**Pull request by SergeyKuryanov**
Fixes #113

Seems like background color of `AVPlayerView` can't be modified, but we can use a trick.
We can add overlay with with a window for video preview. This makes effect of checkerboard background.

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Referenced issues

This pull request has been submitted to:
- [#113: Replace black background on trimming view with a checkerboard NSColor](https://issuehunt.io/repos/119822304/issues/113)
---

IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### CODE REVIEW

File: Gifski/CheckerboardView.swift [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+//  Created by Sergey Kuryanov on 06/10/2019.
+//  Copyright © 2019 Sindre Sorhus. All rights reserved.
+//
+
+import Cocoa
+
+class CheckerboardView: NSView {
+	let gridSize = CGSize(width: 10, height: 10)
+	let firstColor = NSColor.white
+	let secondColor = NSColor.lightGray
```
Comment (sindresorhus) [inline Gifski/CheckerboardView.swift]: This is a bit too dark. I would use `#d1d1d1` (preferably a semantic color close to this).

File: Gifski/CheckerboardView.swift [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	override func draw(_ dirtyRect: NSRect) {
+		super.draw(dirtyRect)
+
+		firstColor.setFill()
+		dirtyRect.fill()
+
+		secondColor.setFill()
+
+		for y in 0...Int(bounds.size.height / gridSize.height) {
+			for x in 0...Int(bounds.size.width / gridSize.width) where x % 2 == y % 2 {
```
Comment (sindresorhus) [inline Gifski/CheckerboardView.swift]: Use this for readability:

```swift
extension BinaryInteger {
	var isEven: Bool { isMultiple(of: 2) }
	var isOdd: Bool { !isEven }
}
```

File: Gifski/CheckerboardView.swift [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+//  Gifski
+//
+//  Created by Sergey Kuryanov on 06/10/2019.
+//  Copyright © 2019 Sindre Sorhus. All rights reserved.
+//
+
+import Cocoa
+
+class CheckerboardView: NSView {
+	private let gridSize = CGSize(width: 10, height: 10)
```
Comment (sindresorhus) [inline Gifski/CheckerboardView.swift]: I would go for 7px.

**Comment by sindresorhus** [comment]
Nice trick!

**Comment by sindresorhus** [comment]
Should we use a dark checkerboard version in dark mode?

**Comment by SergeyKuryanov** [comment]
> Should we use a dark checkerboard version in dark mode?
I think we should. Maybe flip white to black for dark mode?

**Comment by SergeyKuryanov** [comment]
@sindresorhus grid size 7px looks not okay at top, so I set 8px:
Снимок экрана 2019-10-07 в 15 25 31

File: Gifski/util.swift [hunk 4]
```diff
@@ -69,6 +69,14 @@ extension NSColor {
 	}
 }
 
+// MARK: - Checkerboard Colors
+extension NSColor {
+	enum Checkerboard {
+		static let first = NSColor(named: "CheckerboardFirstColor") ?? NSColor.white
+		static let second = NSColor(named: "CheckerboardSecondColor") ?? NSColor.lightGray
+	}
+}
```
Comment (sindresorhus) [inline Gifski/util.swift]: Put this in https://github.com/sindresorhus/Gifski/blob/master/Gifski/Constants.swift and you can just force-unwrap the colors.

**Comment by sindresorhus** [comment]
Make the second color in dark mode `#838485`.

**Comment by sindresorhus** [comment]
Looks great 👌

**Pull request merged**